src/Altinn.App.Core/Features/Correspondence/Builder/CorrespondenceRequestBuilder.cs

```diff
@@ -0,0 +1,299 @@
+using System.Diagnostics.CodeAnalysis;
+using Altinn.App.Core.Features.Correspondence.Exceptions;
+using Altinn.App.Core.Features.Correspondence.Models;
+using Altinn.App.Core.Models;
+
+namespace Altinn.App.Core.Features.Correspondence.Builder;
+
+/// <summary>
+/// Base functionality for correspondence builders
+/// </summary>
+public abstract class CorrespondenceBuilderBase
+{
+    /// <summary>
+    /// Because of the interface-chaining in this builder, some properties are guaranteed to be non-null.
+    /// But the compiler doesn't trust that, so we add this check where needed.
+    ///
+    /// Additionally this method checks for empty strings and empty data allocations.
+    /// </summary>
+    /// <param name="value">The value to assert</param>
+    /// <param name="errorMessage">The error message to throw, if the value was null</param>
+    /// <exception cref="CorrespondenceValueException"></exception>
+    internal static void NotNullOrEmpty([NotNull] object? value, string? errorMessage = null)
```

Comment:
Can we share this function without having `CorrespondenceBuilderBase` as a base class? Is it used for something else? Reason I'm asking is because the symbol is public and so would be part of the surface we have to maintain/can't change

Reply:
Absolutely! I was just hoping to avoid a helper class situation, but that can easily be done 🙂

Reply:
maybe you've already seen it, but in this BCL example it's part of the exception type: https://github.com/dotnet/runtime/blob/6be24fd37e7d9f04c7fa903b8b6912c3eafe7198/src/libraries/System.Private.CoreLib/src/System/ArgumentException.cs#L104
apart from that `ThrowHelper` is also pretty common

Reply:
Ended up with a `BuilderUtils` class. It`s not exposed publicly, so we are free to change that to whatever we want down the line.